select/index.js

```diff
@@ -0,0 +1,313 @@
+class IrnmnSelect extends HTMLElement {
+    constructor() {
+        super();
+        this.selectedOption = null;
+        this.isOpen = false;
+    }
+
+    connectedCallback() {
+        this.setPreselectedOption();  // Set preselected option based on value
+        this.render();
+        this.setupEventListeners();
+    }
+
+    get headingText() {
+        return this.getAttribute('heading-text') || 'Select an Option';
+    }
+
+    get placeholder() {
+        return this.getAttribute('placeholder') || null;
+    }    
+
+    get options() {
+        const optionsAttr = this.getAttribute('options');
+        return optionsAttr ? JSON.parse(optionsAttr) : [];
+    }
+
+    get preselected() {
+        return this.getAttribute('preselected');  // Get the value of the preselected attribute
+    }
+
+    setPreselectedOption() {
+        if (this.preselected) {
+            // Find the index of the option that matches the preselected value
+            const preselectedIndex = this.options.findIndex(
+                (option) => option.value === this.preselected
+            );
+
+            if (preselectedIndex !== -1) {
+                this.selectedOption = preselectedIndex;
+            }
+        }
+    }
+
+    render() {
+        this.innerHTML = `
+            <style>
+                .irnmn-select {
```

Comment:
can you pls add this to an external CSS? 
You have a solution here: https://github.com/ennismore/irnmn-components/pull/5/files#diff-cc99ea6462ffeef52670b9c874ebf5b835899c8bd9cc70eca829ba33b01f2fabR49

Reply:
@samuelsilvapt unfortunately I just couldn't get `loadAndInjectCSS()` to find my css file so am importing css via a js file for now instead. I can refactor this soon but didn't want to waste anymore time.